docs/develop-resources/scenarios/migrate-tf-resource-beta.md

```diff
@@ -1,88 +1,159 @@
-* We require the `dependencies.yaml` to cover all referenced fields, and the `_http.log` showing the Cloud requests. You need to implement those dependencies' MockGCP methods as well.
+*   We require basic test suites and full test suites.
+*   We require two contiguous git-commits reflecting the real GCP and Mock GCP
+    logs. The Mock log should be as much like the real GCP log as possible.
+*   We require two contiguous git-commits reflecting the mutable fields for
+    `update.yaml` and matching changes in the generated
+    `_generated_object_<resource>.golden.yaml` and `_http.log`
+*   We require the `dependencies.yaml` to cover all referenced fields, and the
+    `_http.log` showing the Cloud requests. You need to implement those
+    dependencies' MockGCP methods as well.
 
 ## Add the backward compatible API
 
-Follow [deep-dives Step 2](../deep-dives/2-define-apis.md). We use the Mock GCP and Terraform or DCL based controller to verify the API migration is backward compatible.
+Follow [deep-dives Step 2](../deep-dives/2-define-apis.md). We use the Mock GCP
+and Terraform or DCL based controller to verify the API migration is backward
+compatible.
 
-The PR shall contain the types and deepcopy codes. It shall make modifications to make sure the CRDs are the same, because Beta resource has to be backward compatible. 
+The PR shall contain the types and deepcopy codes. It shall make modifications
+to make sure the CRDs are the same, because Beta resource has to be backward
+compatible.
 
-* You may need to modify the auto-generated `types.go` to keep the existing fields the same (even if it is not following the recommended styles and conventions). You can run `dev/tasks/generate-crds` (repeatedly) to make sure the CRD are the same (comment changes are acceptable).
+*   You may need to modify the auto-generated `types.go` to keep the existing
+    fields the same (even if it is not following the recommended styles and
+    conventions). You can run `dev/tasks/generate-crds` (repeatedly) to make
+    sure the CRD are the same. Comment changes to match the latest protobuf are
+    recommended as long as the content doesn't contain breaking changes. An
+    example of a breaking change would be the comment explicitly says a
+    previously optional field `required`.
 
-* Add `cnrm.cloud.google.com/dcl2crd: "true"` or `cnrm.cloud.google.com/tf2crd: "true"` to the API tag [example](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/0bbac86ace6ab2f4051b574f026d5fe47fa05b75/pkg/controller/direct/redis/cluster/roundtrip_test.go#L92), to continue using DCL-based or Terraform-based controllers.
+*   Add `cnrm.cloud.google.com/dcl2crd: "true"` or
+    `cnrm.cloud.google.com/tf2crd: "true"` to the API tag
+    [example](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/0bbac86ace6ab2f4051b574f026d5fe47fa05b75/pkg/controller/direct/redis/cluster/roundtrip_test.go#L92),
```

Comment:
this is not the right example line right?

Reply:
Nope. Will find a correct example.

Reply:
Done.